File: 0058-length-of-last-word/0058-length-of-last-word.c
int lengthOfLastWord(char* s) {
    int len;
    int i;

    i = 0;
    len = 0;
    while (s[i]){
        i++;
    }
    while (i > 0 && (s[i - 1] == '\t' || s[i - 1] == '\f' || s[i - 1] == ' ' || s[i - 1] == '\n' || s[i - 1] == '\r')) {
        i--;
    }
    while (i > 0 && !(s[i - 1] == '\t' || s[i - 1] == '\f' || s[i - 1] == ' ' || s[i - 1] == '\n' || s[i - 1] == '\r')) {
        len++;
        i--;
    }
    return len;

}

File: 0058-length-of-last-word/0058-length-of-last-word.java
/*
 * @lc app=leetcode id=58 lang=java
 *
 * [58] Length of Last Word
 */

// @lc code=start
class Solution {
    public int lengthOfLastWord(String s) {
        
        int length = 0;
        int i = s.length() - 1;
        
        while(i >= 0 && s.charAt(i) == ' ') {
            i--;
        }
        
        while(i >= 0 && s.charAt(i) != ' ') {
            length++;
            i--;
        }
        
        return length;
    }
}
// @lc code=end

